packages/core/android/src/main/kotlin/com/datadog/reactnative/FrameRateProvider.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative
+
+import com.facebook.react.modules.core.ChoreographerCompat
+
+internal class FrameRateProvider(
+    reactFrameRateCallback: ((Double) -> Unit)?
+) {
+    private val frameCallback: FpsFrameCallback = FpsFrameCallback(reactFrameRateCallback)
+
+    fun start() {
+        frameCallback.reset()
+        frameCallback.start()
+    }
+
+    fun stop() {
+        frameCallback.stop()
+    }
+}
+
+internal class FpsFrameCallback(
+    private val reactFrameRateCallback: ((Double) -> Unit)?
+) :
+    ChoreographerCompat.FrameCallback() {
+    private var mChoreographer: ChoreographerCompat? = null
+
+    private var mLastFrameTime = -1L
+
+    override fun doFrame(time: Long) {
+        if (mLastFrameTime != -1L) {
+            reactFrameRateCallback?.let { it((time - mLastFrameTime).toDouble()) }
```

Comment:
alternatively can be even shorter

```suggestion
            reactFrameRateCallback?.invoke((time - lastFrameTime).toDouble())
```